docs/schemas/trace-analytics/readme.md

```diff
@@ -0,0 +1,91 @@
+
+## Tenets
+
+1. Schemas shall follow [semantic versioning](https://semver.org/), excluding patch version numbers.
+2. Schema versions shall be detached from Data Prepper and Trace Analytics plugin versions.
+3. Index and index template names shall only include the major version (e.g. "otel-v1-apm-span"). The minor version shall be included within the actual schema as a field.
+4. Forward and backward-compatibility promises shall only apply to schemas of the same major version.
+5. A major version increase shall require Data Prepper (writer) artifacts to be made available before Trace Analytics plugin (reader) updates are made available.
+6. A major version increase shall result in a new index template and indexes being created in an Elasticsearch/OpenSearch cluster.
+
+## Versioning Format
+
+A schema will be versioned following the [Semantic Versioning 2.0.0 spec](https://semver.org/). A schema version will include a major and minor version number. Patch version numbers will not be used as "patching" is not applicable to a versioned schema; all changes no matter how trivial will have implications.
+
+1. **Major versions** will be incremented for breaking, backwards-incompatible changes.
+2. **Minor versions** will be incremented for backwards-compatible feature additions. This can be thought of an "append-only" change to the schema.
+
+**Schema versions are detached from Data Prepper and Trace Analytics plugin versions.** A TA plugin version increase does not necessarily affect the version of the schema or Data Prepper. Instead, both DP and TA versions will be *compatible* with a specific schema version. Examples include:
+
+* Trace Analytics plugin v1.5 includes features built on schema version 1.2.0
+* Data Prepper v1.1 emits documents following schema version 1.2.0
+
+### Major version changes
+
+Major version changes include removing a field, renaming a field, or changing an existing field's datatype.
+
+* Schema 1.0 to 2.0 *changes the type of a field* from Keyword to Numeric
+* Schema 1.0 to 2.0 *removes* *field* "latency" from the schema
+* Schema 1.0 to 2.0 *renames field* "end" to "endTime"
+    * A rename is effectively a field addition and removal in a single operation
+
+### Minor version changes
+
+Minor version changes include adding a new field or adding a new nested field.
+
+* Schema 1.2 to 1.3 adds a new field, "fieldC", as a Keyword
+* Schema 2.11 to 2.12 adds a new nested field, "name" to an existing collection, "parentSpan", resulting in "parentSpan.name"
+
+## Compatibility Promises
+
+The following compatibility promises are made *only for schemas of the same major version*.
+
+* ***Backwards compatibility*** - features built on version 1.x of the schema **will not break, but may degrade** if data from a **prior** 1.x schema version is used.
```

Comment:
Suggest specify features as Trace Analytics UI features to avoid confusion, or just reader features according to the context.

Reply:
Counter question: which part is ambiguous, or how did you interpret "features" as? I had assumed that the document context was enough but maybe it wasn't.

Reply:
Upon my first read, I was wondering what "features" include. Data-prepper features or UI features or both.